govcd/any_edgegateway.go

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2022 VMware, Inc.  All rights reserved.  Licensed under the Apache v2 License.
+ */
+
+package govcd
+
+import (
+	"fmt"
+	"net/url"
+
+	"github.com/vmware/go-vcloud-director/v2/types/v56"
+)
+
+// AnyEdgeGateway is a common structure which fetches any type of Edge Gateway (NSX-T or NSX-V)
+// using OpenAPI endpoint. It can be useful to identify type of Edge Gateway or just retrieve common
+// fields - like OwnerRef. There is also a function GetNsxtEdgeGateway to convert it to
+// NsxtEdgeGateway (if it is an NSX-T one)
+type AnyEdgeGateway struct {
+	EdgeGateway *types.OpenAPIEdgeGateway
+	client      *Client
+}
+
+// GetNsxtEdgeGatewayById allows retrieving NSX-T edge gateway by ID for Org admins
+func (adminOrg *AdminOrg) GetAnyEdgeGatewayById(id string) (*AnyEdgeGateway, error) {
+	return getAnyApiEdgeGatewayById(adminOrg.client, id, nil)
+}
+
+// GetNsxtEdgeGatewayById allows retrieving NSX-T edge gateway by ID for Org users
+func (org *Org) GetAnyEdgeGatewayById(id string) (*AnyEdgeGateway, error) {
+	return getAnyApiEdgeGatewayById(org.client, id, nil)
+}
+
+// getNsxtEdgeGatewayById is a private parent for wrapped functions:
+// func (adminOrg *AdminOrg) GetNsxtEdgeGatewayByName(id string) (*NsxtEdgeGateway, error)
+// func (org *Org) GetNsxtEdgeGatewayByName(id string) (*NsxtEdgeGateway, error)
+func getAnyApiEdgeGatewayById(client *Client, id string, queryParameters url.Values) (*AnyEdgeGateway, error) {
+	endpoint := types.OpenApiPathVersion1_0_0 + types.OpenApiEndpointEdgeGateways
+	minimumApiVersion, err := client.getOpenApiHighestElevatedVersion(endpoint)
+	if err != nil {
+		return nil, err
+	}
+
+	if id == "" {
```

Comment:
such check should be first in function :)

Reply:
Moved.